### ISSUE TITLE
IBus 1.5.26 does not recognize Latin American keyboard layouts, due to explicit blacklist.

### ISSUE THREAD

**Issue by sm-programmer** [ibus/ibus#2412]
I've recently stumbled upon a problem, in selecting the Spanish Latin American keyboard layout.
I'm using IBus 1.5.26, on Xubuntu 22.04 (XFCE 4.16).

The layout is not present in the `simple.xml` whitelist, and I had to manually add it in order to use it, which is problematic, since an update in IBus will overwrite the change and has to be applied again.

Then, after checking #2322 and #2153, I discovered commit 568d58d, regarding the implementation of a blacklist, which _explicitly blacklists all Latin American layouts on build time_.

I'd like to know the reason why such a blacklist is in place, and if any steps will be taken to address this issue.
Regards!

**Comment by fujiwarat** [comment]
I don't remember the detail but there would be many the Latin American keymaps but seems 6 keymaps only now.
I can delete xkb:latam:*:* in denylist.txt if you wish to fix it in upstream.
On the other hand I forgot to add denylist.txt in the tarballs.

**Comment by Ark74** [comment]
I've tried applied https://github.com/ibus/ibus/commit/82e38d9e1c06e7fdc0816fe08f4021b8c1cdff10 (also adding the engine/denylist.txt)  to a ubuntu package, recompiled, and seems no Latin American layout is displayed.

Is there some other changed required for LatAm layout to be displayed?
Regards

**Comment by fujiwarat** [comment]
> Is there some other changed required for LatAm layout to be chosen?

You should rebuild ibus/engine/simple.xml.in

**Comment by Ark74** [comment]
@fujiwarat hello again!

Sorry, I'm a bit stranger to ibus, could you point me in the right direction as in how is the simple.xml.in rebuild, I thought it would be done while binaries were building, now I know that's not the case.
Regards

**Comment by fujiwarat** [comment]
> could you point me in the right direction as in how is the simple.xml.in rebuild

```
% ./configure
% cd engine
% make simple.xml.in
```

simple.xml.in is updated before ibus tarball will be released but not now.

**Comment by Ark74** [comment]
@fujiwarat thank you, for some reason I keep getting the same result.
I'll wait on the next release, again thank you very much.
